qctoolkit/serialization.py

```diff
@@ -141,6 +150,21 @@ def delete(self, identifier):
         except FileNotFoundError as fnf:
             raise KeyError(identifier) from fnf
 
+    def list_contents(self) -> Set[str]:
+        contents = set()
+        for dirpath, dirs, files in os.walk(self._root):
```

Comment:
`os.walk` goes through the complete tree. We only return top level jsons in `FilesystemBackend.get`

Reply:
Which is why there's a line 159 which aborts the iteration after the first level. The reason to still use os.walk() was that it conveniently separates dirs and files and gives me the base dirpath as well in one single call.